DEHPEcosimPro/ViewModel/Rows/ChooseMappingColumnsViewModel.cs

```diff
@@ -0,0 +1,170 @@
+// 
+//    The DEHPEcosimPro is distributed in the hope that it will be useful,
+//    but WITHOUT ANY WARRANTY; without even the implied warranty of
+//    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+//    Lesser General Public License for more details.
+// 
+//    You should have received a copy of the GNU Lesser General Public License
+//    along with this program; if not, write to the Free Software Foundation,
+//    Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+// </copyright>
+// --------------------------------------------------------------------------------------------------------------------
+
+namespace DEHPEcosimPro.ViewModel.Rows
+{
+    using System.Collections.Generic;
+    using System.Linq;
+
+    using CDP4Common.EngineeringModelData;
+    using CDP4Common.SiteDirectoryData;
+
+    using DEHPCommon.UserInterfaces.Behaviors;
+    using DEHPCommon.UserInterfaces.ViewModels.Interfaces;
+
+    using DevExpress.Mvvm.Native;
+
+    using ReactiveUI;
+
+    /// <summary>
+    /// ViewModel of the dialog to match columns of parameter to a variable
+    /// </summary>
+    public class ChooseMappingColumnsViewModel : ReactiveObject, ICloseWindowViewModel
+    {
+        /// <summary>
+        /// ViewModel of the dialog to match columns of parameter to a variable
+        /// </summary>
+        public ChooseMappingColumnsViewModel()
+        {
+        }
+
+        /// <summary>
+        /// ViewModel of the dialog to match columns of parameter to a variable
+        /// </summary>
+        public ChooseMappingColumnsViewModel(VariableBaseRowViewModel variable, ParameterOrOverrideBase parameter)
+        {
+
+            if (parameter.ParameterType is SampledFunctionParameterType parameterType)
+            {
+                this.ListOfParameterToMatch.AddRange(parameterType.IndependentParameterType.Select(x => x.ParameterType.ShortName));
+                this.ListOfParameterToMatch.AddRange(parameterType.DependentParameterType.Select(x => x.ParameterType.ShortName));
+            }
+
+            if (variable is ArrayVariableRowViewModel array)
+            {
+                this.VariableName = array.Name;
+                this.IsList = array.IsList;
+                List<IGrouping<string, VariableRowViewModel>> groupby;
+
+                if (array.IsList)
+                {
+                    groupby = array.Variables.GroupBy(x => x.Name).ToList();
+                }
+                else
+                {
+                    groupby = array.Variables.GroupBy(x => this.GetColumnNumberFromValueName(x.Reference.DisplayName.Text)).ToList();
+                }
+
+                foreach (var group in groupby)
+                {
+                    this.ListOfVariableToMap.Add(new ChooseMappingRowsViewModel(group.ToList(), array.IsList));
+                }
+            }
+
+            this.ParameterName = ModelCode(parameter);
```

Comment:
missing this, also why not using the Methods from the type itself:  parameter.ModelCode();